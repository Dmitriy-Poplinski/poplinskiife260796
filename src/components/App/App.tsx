import React, { useEffect, useState } from 'react';

import { Footer } from '../Footer/Footer';
import { Header } from '../Header/Header';
import { Main } from '../Main/Main';

import classes from './App.module.css';

export function App() {
    const [ images, setImages ] = useState([
        'header_cover',
        'left_panel_cover',
        'main_body_cover',
        'right_panel_cover',
        'footer_cover'
    ]);

    const shuffleImageArray = () => {
        const newImages = images.sort(() => Math.random() - 0.5);
        setImages([...newImages]);
    };

    const main_images = [images[1], images[2], images[3]];

    useEffect(() => {
        const time = +(0.52 * 60 * 1000).toFixed(0);

        setInterval(shuffleImageArray, time);
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])

    return (
        <div className={classes.app} data-testid="app">
            <Header image={images[0]}/>
            <Main images={main_images} shuffleImageArray={shuffleImageArray}/>
            <Footer image={images[4]}/>
        </div>
    );
}
