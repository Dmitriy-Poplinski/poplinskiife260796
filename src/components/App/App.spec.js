import { render, screen } from '@testing-library/react';

import App from './App';

describe('App:', () => {
    test('App snapshot', () => {
        render(<App />);

        expect(screen.getByTestId('app')).toMatchSnapshot();
    });
});
