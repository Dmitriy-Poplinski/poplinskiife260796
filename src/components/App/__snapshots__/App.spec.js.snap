// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App: App snapshot 1`] = `
<div
  class="app"
  data-testid="app"
>
  <div
    class="header__main"
    data-testid="header"
    style="background: url(header_cover.jpg);"
  >
    Header
  </div>
  <div
    class="main"
    data-testid="main"
  >
    <div
      class="left_panel"
      data-testid="left_panel"
      style="background: url(left_panel_cover.jpg);"
    >
      Left Panel
    </div>
    <div
      class="main_body"
      data-testid="main_body"
      style="background: url(main_body_cover.jpg);"
    >
      <button
        class="button"
        data-testid="button"
      >
        Click me
      </button>
    </div>
    <div
      class="right_panel"
      data-testid="right_panel"
      style="background: url(right_panel_cover.jpg);"
    >
      Right Panel
    </div>
  </div>
  <div
    class="footer__main"
    data-testid="footer"
    style="background: url(footer_cover.jpg);"
  >
    Footer
  </div>
</div>
`;
